worker/src/request_init.rs

```diff
@@ -281,49 +297,134 @@ pub struct MinifyConfig {
 
 /// Configuration options for Cloudflare's image optimization feature:
 /// <https://blog.cloudflare.com/introducing-polish-automatic-image-optimizati/>
-#[wasm_bindgen]
-#[derive(Clone, Copy)]
+#[derive(Clone, Copy, Default, Serialize)]
+#[serde(rename_all = "kebab-case")]
 pub enum PolishConfig {
+    #[default]
     Off,
     Lossy,
     Lossless,
 }
 
-impl Default for PolishConfig {
-    fn default() -> Self {
-        Self::Off
-    }
+#[derive(Clone, serde::Serialize)]
+#[serde(untagged)]
+pub enum ResizeBorder {
+    Uniform {
+        color: String,
+        width: usize,
+    },
+    Varying {
+        color: String,
+        top: usize,
+        right: usize,
+        bottom: usize,
+        left: usize,
+    },
 }
 
-impl From<PolishConfig> for &str {
-    fn from(conf: PolishConfig) -> Self {
-        match conf {
-            PolishConfig::Off => "off",
-            PolishConfig::Lossy => "lossy",
-            PolishConfig::Lossless => "lossless",
-        }
-    }
+/// Configuration options for Cloudflare's image resizing feature:
+/// <https://developers.cloudflare.com/images/image-resizing/>
+#[derive(Clone, Default, serde::Serialize)]
+pub struct ResizeConfig {
+    pub anim: Option<bool>,
+    pub background: Option<String>,
+    pub blur: Option<u8>,
+    pub border: Option<ResizeBorder>,
+    pub brightness: Option<f64>,
+    pub compression: Option<ResizeCompression>,
+    pub contrast: Option<f64>,
+    pub dpr: Option<f64>,
+    pub fit: Option<ResizeFit>,
+    pub format: Option<ResizeFormat>,
+    pub gamma: Option<f64>,
+    pub gravity: Option<ResizeGravity>,
+    pub height: Option<usize>,
+    pub metadata: Option<ResizeMetadata>,
+    pub onerror: Option<ResizeOnerror>,
+    pub quality: Option<usize>,
+    pub rotate: Option<usize>,
+    pub sharpen: Option<usize>,
+    pub trim: Option<ResizeTrim>,
+    pub width: Option<usize>,
+}
+
+#[derive(Clone, Copy, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum ResizeCompression {
+    Fast,
+}
+
+#[derive(Clone, Copy, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum ResizeFit {
+    ScaleDown,
+    Contain,
+    Cover,
+    Crop,
+    Pad,
+}
+
+#[derive(Clone, Copy, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum ResizeFormat {
```

Comment:
There's also `jpeg` and `baseline-jpeg` https://developers.cloudflare.com/images/image-resizing/resize-with-workers#format